[docs] [PLAT-8479] added a note for installing YBC during DB version upgrades. (#17019)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md

The YugabyteDB release that is powering a universe can be upgraded to get new features and fixes included in the release.

Before starting the upgrade:

- Ensure that the YB Controller port 18018 is open on all YugabyteDB nodes.

- Consider importing a specific YugabyteDB release into YugabyteDB Anywhere, as follows:

  - Click the user profile icon and select **Releases**.

  - Click **Import** to open the **Import Releases** dialog shown in the following illustration:

    ![Import Releases](/images/yp/import-releases.png)

  - Specify the release version to import.

>     {{< note title="YugabyteDB upgrades for version 2.16 and later" >}}

Reviewer: ```suggestion
    {{< note title="Manually-provisioned on-premises universe upgrades" >}}
```

> For v2.16 and later, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) universe.

Reviewer 2: @nbhatia-yb,, As we don't have the fix in 2.16.3, so Akki mentioned that we should have the workaround mentioned until we release 2.16.4.
We have a separate ticket for docs changes related to manually provisioned systemd managed universe [PLAT-8123]
Also, @aishwarya24 This workaround is for manually provisioned cron managed universes. We don' support ybc installation on manually provisioned systemd managed universe during software upgrade until we release 2.16.4 or 2.18.

> For v2.16 and later, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) universe.

Reviewer 3: @nbhatia-yb @aishwarya24 This workaround is for all on-prem manually provisioned universes in 2.16.3. Also, this would not work if there are read replicas. We should probably be explicit about this in our latest 2.16 docs till we have our docs and bug fixes as part of 2.16.4 and 2.18.

> For v2.16 and later, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) universe.

Author: @vpatibandla-yb - I have made the note specific to the version 2.16.3 and have included it for preview(2.17) and stable(2.16) docs. Could you please take a look and let me know if I got it right.

>   YB Controller does not get installed during a YugabyteDB software upgrade for an [on-prem](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) manually provisioned universe for versions 2.16 and later.

Reviewer: ```suggestion
For v2.16 and later, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) universe.
```

  The workaround is to explicitly install the YB Controller by calling the following API after the software upgrade:

  ```sh

>   curl --location --request PUT '<YBA-url>/api/v1/customers/<customerUUID>/universes/<universeUUID>/ybc/install' \

Author: @vpatibandla-yb . Sure. But the only way I can think of is probably by asking the customer to go to network tab in the web inspector to get the UUID details and others . This does not seem like an appropriate suggestion (maybe a security issue) . Also, are there other straightforward ways to fetch these ID's ?

    {{< note title="Manually-provisioned on-premises universe upgrades" >}}

>   For v2.16.3, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) cron managed universe.

Reviewer: ```suggestion
  For v2.16.3, YB Controller does not get installed during a YugabyteDB software upgrade of a manually provisioned [on-premises](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises/) cron-managed universe.
```
